src/Events/VerbsStateInitialized.php

```diff
@@ -14,15 +15,18 @@ public function __construct(
         public int|string $state_id,
         public string $state_class,
         public array $state_data,
-        public bool $singleton = false,
     ) {}
 
     /** @return StateCollection<int, TStateType> */
     public function states(): StateCollection
     {
-        return StateCollection::make([
-            $this->singleton ? $this->state_class::singleton() : $this->state_class::load($this->state_id),
-        ]);
+        $state = is_subclass_of($this->state_class, SingletonState::class)
```

Comment:
Why are you sometimes using `is_subclass_of` and other times using `instanceof`?

Reply:
`is_subclass_of` lets you compare strings, whereas `instanceof` checks an object.